src/Microbuild.Settings.targets

```diff
@@ -155,9 +156,18 @@
                 throw new ArgumentException("Language " + Language + " is not supported");
         }
         
-        AssemblyFileVersionPath = Path.GetTempFileName() + "." + Language;
-        Log.LogMessage("Generating AssemblyFileVersion file: " + AssemblyFileVersionPath, MessageImportance.High);
-        File.WriteAllText(AssemblyFileVersionPath, stringBuilder.ToString());
+        var fileContents = stringBuilder.ToString();
+        AssemblyFileVersionPath = Path.Combine(Path.GetTempPath(), string.Join(".", AssemblyName, "AssemblyFileVersion", Language));
+        
+        if (File.Exists(AssemblyFileVersionPath) && File.ReadAllText(AssemblyFileVersionPath) == fileContents)
```

Comment:
I don't think we do any direct I/O in the targets / settings files.  I doubt this would cause an issue though.